indy-wrapper/src/main/java/nl/quintor/studybits/indy/wrapper/Issuer.java

```diff
@@ -2,114 +2,99 @@
 
-        return Ledger.buildSchemaRequest(issuerDid, schemaDefinition.toJSON())
+        return Ledger.buildSchemaRequest(getIssuerDid(), createSchemaResult.getSchemaJson())
                      .thenCompose(wrapException(request -> {
                          log.debug("{}: Submitting buildSchema request {}", this.name, request);
-                         return signAndSubmitRequest(request, issuerDid);
+                         return signAndSubmitRequest(request, getIssuerDid());
                      }))
-                     .thenApply(requestResponse -> schemaKey);
+                     .thenApply(requestResponse -> createSchemaResult.getSchemaId());
     }
 
-    public CompletableFuture<String> defineClaim(SchemaKey schemaKey) throws JsonProcessingException, IndyException {
-        log.debug("{}: Defining claim for schemaKey {}", name, schemaKey);
-        return getSchema(issuerDid, schemaKey)
+    public CompletableFuture<String> defineCredential(String schemaId) throws JsonProcessingException, IndyException {
+        log.debug("{}: Defining credential for schemaId {}", name, schemaId);
+        return getSchema(getIssuerDid(), schemaId)
                 .thenCompose(wrapException(schema ->
-                        Anoncreds.issuerCreateAndStoreClaimDef(wallet.getWallet(), issuerDid, schema.toJSON(), "CL", false)))
-                .thenCompose(wrapException(claimDefJson -> {
-                    log.debug("{}: got claim def json {}", name, claimDefJson);
-                    ClaimDefinition claimDefinition = JSONUtil.mapper.readValue(claimDefJson, ClaimDefinition.class);
-                    log.debug("{}: building claim def txn with submitterDid {} xref {} signatureType {} data {}", name, issuerDid, claimDefinition.getRef(), claimDefinition.getSignatureType(), claimDefinition.getData().toString());
-                    return Ledger.buildClaimDefTxn(issuerDid, claimDefinition.getRef(), claimDefinition.getSignatureType(), claimDefinition.getData().toString());
-                })).thenCompose(wrapException(claimDefTxn -> {
-                    log.debug("{} Signing and sending claimDefTx: {}", name, claimDefTxn);
-                    return Ledger.signAndSubmitRequest(pool.getPool(), wallet.getWallet(), issuerDid, claimDefTxn);
-                })).thenApply((response) -> {
-                            log.debug("{} Got ClaimDefTxn response: {}", name, response);
-                            return response;
-                        }
-                );
+                {
+                    return Anoncreds.issuerCreateAndStoreCredentialDef(wallet.getWallet(), getIssuerDid(), schema.toJSON(), name + "_" + schemaId, "CL", "{\"support_revocation\":false}");
+                }))
+                .thenCompose(wrapException(createAndStoreCredentialDefResult -> {
+                    return Ledger.buildCredDefRequest(getIssuerDid(), createAndStoreCredentialDefResult.getCredDefJson())
```

Comment:
add debug loggers here. We always have problems getting information about what goes wrong in the futures here, so debug loggers are crucial!

Reply:
In particular, I mean add a debug log for the input parameter 'createAndStoreCredentialDefResult'